test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/TestSuitInitializer.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common;
+
+/* Responsible for all initilisation before actual test startup -
+* Creation of Azure resources - Eventhubs and Storage Queues
+* Function App startup
+*/
+public class TestSuitInitializer
+{
+	private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+	public void InitializeTestSuit(Language language, BrokerType brokerType)
+	{
+		CreateAzureResources(language, brokerType);
+		Task.WaitAll(StartupApplicationAsync(language, brokerType));
+	}
+
+	private void CreateAzureResources(Language language, BrokerType brokerType)
+	{
+		var taskList = new List<Task>();
+
+		if (BrokerType.EVENTHUB == brokerType)
+		{
+			taskList.Add(CreateEventHubAsync(language));
+		}
+
+		taskList.Add(ClearStorageQueueAsync(language, brokerType));
+
+		Task.WaitAll(taskList.ToArray());
+	}
+
+	private async Task StartupApplicationAsync(Language language, BrokerType brokerType)
+	{
+		IExecutableCommand<Process> command =
+			ShellCommandFactory.CreateShellCommand(ShellCommandType.DOCKER_RUN, brokerType, language);
+		IExecutor<IExecutableCommand<Process>, Process> executor = new ShellCommandExecutor();
+		ProcessLifecycleManager.GetInstance().AddProcess(await executor.ExecuteAsync(command));
+	}
+
+	private async Task ClearStorageQueueAsync(Language language, BrokerType brokerType)
```

Comment:
I think we can simplify such implementations by having array of changing variables and then using foreach. That way the algorithm can be immutable, only array(proably global) need to change for future iterations.

Reply:
Sure. Will take it up in the improvement task.